contrib_bots/lib/poll.py

```diff
@@ -0,0 +1,103 @@
+class PollHandler(object):
+    '''
+    This plugin allows users to create polls.
+    '''
+
+    def __init__(self):
+        # polls is a dict of dict. First dict corresponds to poll title,
+        # second refers to answers in poll
+        self.polls = dict()
+
+        # streams is a convenience object that keeps track of the latest
+        # question for each stream.
+        self.streams = dict()
+
+    def usage(self):
+        return '''
+        This plugin facilitates polling.
+
+        Commands:
+        @poll init <question>: <option1>, <option2>, <option3>
+        A valid poll must use a colon (:) to divide question from answers.
+        Answers must be seperate by coma (,).
+
+        @poll <option1>
+        Question will be infered by scanning channel for last poll with
+        valid answer.
+
+        @poll result poll title
+        Returns a formatted version of the poll.
+
+        @poll help
+        Returns usage for polling bot.
+
+        >>> @poll init what's the best ice cream flavour: vanilla, chocolate
+        <<< Question successfully created! Answer away
+        >>> @poll vanilla
+        <<< vanilla has 1 vote
+        >>> @poll result what's the best ice cream flavour
```

Comment:
Hm, what do you think about doing something like `@poll result current` to get the result for the current poll? This would also entail restricting the keyword 'current' from being initialized because we can't have conflicting names (though I don't see this as an issue since I don't think anyone would make a poll called current...). Otherwise, user's can still do `@poll result <question>` either to retrieve previous polls or their current poll (at the risk of typos).